scripts/west_commands/sbom/license_utils.py

```diff
@@ -0,0 +1,140 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+'''
+Utility functions for handling licenses and license expressions.
+'''
+
+import re
+from pathlib import Path
+import yaml
+from data_structure import DataBaseClass, License
+
+
+spdx_licenses: 'dict(License)' = dict()
+license_texts: 'list(License)' = list()
+license_by_id: 'dict(License)' = dict()
+
+
+def is_spdx_license(id: str):
+    '''Returns True if id is in SPDX license list.'''
+    return id.upper() in spdx_licenses
+
+
+def get_license(id: str) -> 'License|None':
+    '''Returns license information based on provided id.'''
+    id = id.upper()
+    if id in license_by_id:
+        return license_by_id[id]
+    return None
+
+
+def get_license_texts() -> 'list[License]':
+    '''Return list of all licenses from "license-texts.yaml" file'''
+    return license_texts
+
+
+LICENSE_EXPR_RE = re.compile(r'\s*([a-z0-9\.\-\:]+|\+|\)|\()\s*', re.IGNORECASE)
+
+
+def tokenize_license_expr(expr: str) -> 'list[str]|None':
+    '''Split license expression string into list of expression tokens.'''
+    result = list()
+    expr = expr.strip()
+    while len(expr) > 0:
+        m = LICENSE_EXPR_RE.match(expr)
+        if m is None:
+            return None
+        result.append(m.group(1))
+        expr = expr[m.end():]
+    return result
+
+
+class SPDXLicenseExprInfo(DataBaseClass):
+    '''License expression information parsed by get_spdx_license_expr_info.
+
+    Attributes:
+        expr          Original normalized expression string
+        friendly_expr User friendly version of the expression
+        valid         True if expression was correctly parsed
+        is_id_only    True if the expression contains actually just a single id
+        licenses      Set of normalized license identifiers used in this expression
+        or_present    True if 'OR' operator is in the expression
+    '''
+    expr: str
+    friendly_expr: str
+    valid: bool
+    is_id_only: bool = False
+    licenses: 'set[str]' = set()
+    or_present: bool = False
+
+
+def get_spdx_license_expr_info(expr: str) -> SPDXLicenseExprInfo:
```

Comment:
Are those class methods or just standalone functions?

Reply:
The intention was standalone function that returns multiple values. I used class to enclose them in a single, well defined container.